src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
Feed multiple pipes, or just one and let it dispatch to multiple if so configured?